tests/test_datetime.py

```diff
@@ -827,3 +827,30 @@ def test_read_timezone_geojson(path_test_tz_geojson):
     with fiona.open(path_test_tz_geojson) as c:
         items = list(c)
         assert items[0]["properties"]["test"] == "2015-04-22T00:00:00+07:00"
+
+
+def test_property_setter_lookup(tmp_path):
+    """Demonstrate fix for #1376."""
+
+    class MyDate(datetime.date):
+        pass
+
+    feat = Feature.from_dict(
+        {
+            "properties": {"when": MyDate(2024, 4, 15)},
```

Comment:
@jorisvandenbossche I believe this is for testing purposes equivalent to the situation with `pd.Timestamp`, yes?

Reply:
Yes, that should cover that use case. Thanks!